adot-testbed/app/src/test/java/software/amazon/adot/testbed/LogsTests.java

```diff
@@ -0,0 +1,169 @@
+    }
+
+    @Test
+    void testJson() throws Exception {
+        String logStreamName = "json-logstream-" + uniqueID;
+        collector = createAndStartCollector("/configurations/config-json.yaml", "/logs/testingJSON.log", logStreamName);
+
+        validateLogs(logStreamName , "/logs/testingJSON.log");
+        collector.stop();
+    }
+
+    void validateLogs(String testLogStreamName, String logFilePath) throws Exception {
+        var file = new File(logFilePath);
+        var lines = new HashSet<String>();
+
+        try (InputStream inputStream = getClass().getResourceAsStream(logFilePath);
+             BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                lines.add(line);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Error reading from the file: " + logFilePath, e);
+        }
+
+        var cwClient = CloudWatchLogsClient.builder()
+            .build();
+
+        var objectMapper = new ObjectMapper();
+
+        RetryerBuilder.<Void>newBuilder()
+            .retryIfException()
+            .retryIfRuntimeException()
+            .retryIfExceptionOfType(org.opentest4j.AssertionFailedError.class)
+            .withWaitStrategy(WaitStrategies.fixedWait(10, TimeUnit.SECONDS))
+            .withStopStrategy(StopStrategies.stopAfterAttempt(5))
+            .build()
+            .call(() -> {
+                var now = Instant.now();
+                var start = now.minus(Duration.ofMinutes(2));
+                var end = now.plus(Duration.ofMinutes(2));
+                var response = cwClient.getLogEvents(GetLogEventsRequest.builder().logGroupName("adot-testbed/logs-component-testing/logs")
+                    .logStreamName(testLogStreamName)
+                    .startTime(start.toEpochMilli())
+                    .endTime(end.toEpochMilli())
+                    .build());
+
+                var events = response.events();
+                var receivedMessages = events.stream().map(x -> x.message()).collect(Collectors.toSet());
+
+                // Extract the "body" field from each received message that is received from cloudwatch in JSON Format
+                var messageToValidate = receivedMessages.stream()
+                    .map(message -> {
+                        try {
+                            JsonNode jsonNode = objectMapper.readTree(message);
+                            return jsonNode.get("body").asText();
+                        } catch (Exception e) {
+                            return null;
+                        }
+                    })
+                    .filter(Objects::nonNull)
+                    .collect(Collectors.toSet());
+
+                //Validate body field in JSON-messageToValidate with actual log line from the log file.
+                assertThat(messageToValidate.containsAll(lines)).isTrue();
```

Comment:
Do we need to do an inverse assertion also? 
```suggestion
                assertThat(messageToValidate.containsAll(lines)).isTrue();
                assertThat(lines.containsAll(messageToValidate)).isTrue();
```

Reply:
We could also validate that their size is the same. This would ensure that no extra logs were went. We expect 1:1 correct?

Reply:
yes that's correct it can also be like - assertThat(receivedMessages).containsExactlyInAnyOrderElementsOf(lines); added assertion.